### DISCUSSION TITLE
Can I use T2 not T2 FLAIR?

### DISCUSSION THREAD

**Discussion by Minarose** [McIntosh-Lab/tvb-ukbb#175] [Help] [answered]
Hello 👋🏻 

I have T2w images that I'd like to include as input, however, I noticed that the patterns_actions (line 702) and directories created (line 684) in the [bb_file_manager.py](https://github.com/McIntosh-Lab/tvb-ukbb/blob/main/bb_pipeline_tools/bb_file_manager.py#L709-L735) only include T2 FLAIR. Is there a reason why just T2 is not included or can we go ahead and modify it to include *T2w*.nii.gz?

**Comment by shen4brains** [comment, marked as answer]
Hi Minarose,

T2 images can be handled by modifying, as you say, the expressions in [bb_file_manager.py](https://github.com/McIntosh-Lab/tvb-ukbb/blob/3b5f763475c27f93921842c5a1e6d8cb49afa14e/bb_pipeline_tools/bb_file_manager.py#L706). There's an example of T2w (not FLAIR) processing in the `Cam-CAN` branch. See the code block starting [here](https://github.com/McIntosh-Lab/tvb-ukbb/blob/98a7d7a4ff8d26fd828f6f3e7070150f5cb13e01/bb_structural_pipeline/bb_struct_init#L183) for how that branch handles T2w processing.

**Reply by Minarose** [reply]
Perfect, thank you!

**Comment by justinliuyuwang** [comment]
Hi Minarose, it looks like you're using the main branch. I've recently made some important changes to the pipeline for generating correct TVB inputs. Please be sure to pull the most recent version of the repository before running your data!

**Reply by Minarose** [reply]
Thanks for letting me know!
